manifests/install.pp

```diff
@@ -4,6 +4,8 @@
 class candlepin::install {
   assert_private()
 
+  ensure_packages([$candlepin::java_package])
+
   package { ['candlepin']:
     ensure => $candlepin::version,
```

Comment:
You should ensure it's installed prior to candlepin:

```suggestion
    ensure => $candlepin::version,
    require => Package[$candlepin::java_package],
```

This is because Candlepin requires java:
```console
$ rpm -q --requires candlepin-3.1.22-1.el7.noarch.rpm
warning: candlepin-3.1.22-1.el7.noarch.rpm: Header V4 RSA/SHA256 Signature, key ID e0745924: NOKEY
/usr/bin/env
/usr/bin/python
config(candlepin) = 3.1.22-1.el7
java >= 1:1.8.0
javapackages-tools
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
tomcatjss >= 7.2.1-7.1
wget
```

Otherwise you may end up with both a Java 8 and Java 11 installed if Java 11 would suffice. That is assuming `$java_package` provides `java` of course.

Reply:
Good observation - this very thing happened in my testing!

Reply:
Hmm. What's the right way to make this look now given recent updates?

Reply:
Inside `if $candlepin::java_package {}` you can add `Package[$candlepin::java_package] -> Package['candlepin']`.

Reply:
Thanks! Updated